static/src/js/containers/DatepickerContainer/DatepickerContainer.jsx

```diff
@@ -35,13 +36,33 @@ class DatepickerContainer extends Component {
 
   /**
    * Set view back to the default when a user closes the datepicker
+   * Also set the datetime to be the endOf or startOf day depending on the type
    */
-  onBlur() {
-    const { viewMode } = this.props
+  onBlur(value) {
+    const {
+      format,
+      onSubmit,
+      type,
+      viewMode
+    } = this.props
 
     this.picker.current.setState({
       currentView: viewMode
     })
+
+    let valueToSet = null
+
+    if (typeof value !== 'string' && moment.isMoment(value) && !isCustomTime(value)) {
```

Comment:
Could this accidentally set the `onSubmit(null)` if the conditional fails?

Reply:
should be resolved